packages/perennial/contracts/multiinvoker/MultiinvokerRollup.sol

```diff
@@ -0,0 +1,348 @@
+                (product, amount, ptr) = _decodeAddressAmount(input, ptr);
+
+                _openMakeFor(IProduct(product), amount);
+            } else if (action == 6) { // CLOSE_MAKE
+                address product; UFixed18 amount;
+                (product, amount, ptr) = _decodeAddressAmount(input, ptr);
+
+                _closeMakeFor(IProduct(product), amount);
+            } else if (action == 7) { // CLAIM 
+                address product; uint256[] memory programIds;
+                (product, programIds, ptr) = _decodeProductPrograms(input, ptr);
+
+                _claimFor(IProduct(product), programIds);
+            } else if (action == 8) { // WRAP 
+                address receiver; UFixed18 amount;
+                (receiver, amount, ptr) = _decodeAddressAmount(input, ptr);
+
+                _wrap(receiver, amount);
+            } else if (action == 9) { // UNWRAP
+                address receiver; UFixed18 amount;
+                (receiver, amount, ptr) = _decodeAddressAmount(input, ptr);
+
+                _unwrap(receiver, amount);
+            } else if (action == 10) { // WRAP_AND_DEPOSIT
+                address account; address product; UFixed18 amount;
+                (account, product, amount, ptr) = _decodeAddressAddressAmount(input, ptr);
+                
+               _wrapAndDeposit(account, IProduct(product), amount);
+            } else if (action == 11) { // WITHDRAW_AND_UNWRAP
+                address receiver; address product; UFixed18 amount;
+                (receiver, product, amount, ptr) = _decodeAddressAddressAmount(input, ptr);
+
+                _withdrawAndUnwrap(receiver, IProduct(product), amount);
+            } else if (action == 12) { // VAULT_DEPOSIT
+                address depositer; address vault; UFixed18 amount;
+                (depositer, vault, amount, ptr) = _decodeAddressAddressAmount(input, ptr);
+
+                _vaultDeposit(depositer, IPerennialVault(vault), amount);
+            } else if (action == 13) { // VAULT_REDEEM
+                address vault; UFixed18 shares;
+                (vault, shares, ptr) = _decodeAddressAmount(input, ptr);
+
+                _vaultRedeem(IPerennialVault(vault), shares);
+            } else if (action == 14) { // VAULT_CLAIM
+                address owner; address vault;
+                (owner, vault, ptr) = _decodeAddressAddress(input, ptr);
+
+                _vaultClaim(IPerennialVault(vault), owner);
+            } else if (action == 15) { // VAULT_WRAP_AND_DEPOSIT 
+                address account; address vault; UFixed18 amount;
+                (account, vault, amount, ptr) = _decodeAddressAddressAmount(input, ptr);
+
+                _vaultWrapAndDeposit(account, IPerennialVault(vault), amount);
+            }
+        }
+    }
+
+    /// @notice Unchecked sets address in cache
+    /// @param addr Address to add to cache 
+    function _setAddressCache(address addr) private {
```

Comment:
naming: we use the term `account` throughout our codebase instead of `address` to solving name collision issues with the `address` type keyword and avoid having to use shortenings like `addr`.

You might want to change up the naming for your address cache and surrounding code to refer to these as `account` instead.

Note that stuff like `_decodeAddress` might still make sense as-is though, since these refer to the actual type.

Reply:
address cache is for vaults, products, etc too tho so the general storing of the type

Reply:
unless we also do the above change i guess